include/exadg/fluid_structure_interaction/precice/driver_fluid.h

```diff
@@ -218,7 +216,32 @@ class DriverFluid : public Driver<dim, Number>
 
       // post-solve
       if(is_new_time_window)
+      {
+        // computes new time-step size
         fluid->time_integrator->advance_one_timestep_post_solve();
+        // next, we synchronize the time-step sizes. Subcycling would in theory be compatible in
+        // explicit coupling schemes here. In implicit coupling schemes, we need matching
+        // time-window sizes (either constant (serial or parallel schemes) or adaptively (serial
+        // scheme)) as the time-step size push back happens in the
+        // 'advance_one_timestep_post_solve()' and we cannot change two subsequent time-step sizes
+        // without a push back operation, otherwise we falsify the time integrator. However, the
+        // treatment of the boundary conditions is not handled correctly when using subcycling, as
+        // we apply the 'whole displacement' within the first time-step. In order to cope with
+        // subcycling, either the boundary condition needs to be adopted or we need to wait for a
+        // newer preCICE version (currently v2.3.0) which can handle this.
+        // Hence, we do not adjust the time-step-size of the fluid solver, but rather Assert that it
+        // is still valid. Note that we would run into a preCICE error otherwise reporting the same
+        // issue otherwise.
```

Comment:
This text is hard to understand in my opinion. The documentation of `Driver` (either the base class or both derived classes, probably the base class) has to describe certain nomenclature in my opinion, e.g.

time window: denotes time interval after which FSi coupling takes place
subcycling: a single field solver performs several time steps until the next FSI coupling time window is reached
time step size: refers to single field solvers. In case of subcycling, one time window consists of several time step sizes.

I don't know if this is a precise description (I only want to show which type of information is necessary in my opinion).

Next, describe how the time window size is determined, since this probably comes from pecice. Next, also describe whether the single field solvers have to hit the begin/end of a time window exactly, or whether the time step sizes between fluid and the precice time-window (as well as solid and the precice time-window) may be staggered. Then, assuming matching time windows, may the solid and fluid time step sizes differ in case of subcycling?

I don't know whether it make sense to discuss here constant vs. adaptive schemes (in terms of precice and the window, if I understand it correctly?)? The implementation updates `this->allowed_time_step_size`, so it looks like the implementation is capable of adaptive time window sizes?

I do not understand explicit vs. implicit coupling schemes. What do you use currently? I would have expected that an implicit coupling scheme is used.

I think this is not the right place to discuss serial vs. parallel schemes (again, this would need precise definitions of serial/parallel so that it is understandable from an ExaDG perspective).

Reply:
Ok, I can clear up things, but it might become lengthy. Should I still put it in this place as a source code comment?

Reply:
One could also limit the scope, so that one does not have to explain everything and that it does not become overly lengthy. For example, if explicit vs. implicit is a detail / configuration option of precice, no need to discuss this here.